docs/DeveloperGuide.md

```diff
@@ -1,34 +1,148 @@
 # Developer Guide
+* [Setting up](#1-setting-up)
+* [Design](#2-design)
+* [Implementation](#3-implementation)
+  * [Undo features](#31-proposed-undo-feature)
+    * [View By Category](#311-view-by-category)
+* [Documentation](#4-documentation)
+* [Testing](#5-testing)
+* [Dev Ops](#6-dev-ops)
+  * [Appendix A: Product Scope](#61-appendix-aproduct-scope)
+  * [Appendix B:Value proposition](#62-appendix-bvalue-proposition)
+  * [Appendix C:User Stories](#63-appendix-cuser-stories)
+  * [Appendix D:Non-Functional Requirements](#64-appendix-dnon-functional-requirements)
+  * [Appendix E:Glossary](#65-appendix-eglossary)
+  * [Appendix F:Instructions for Manual Testing](#66-appendix-finstructions-for-manual-testing)
+## 1. Setting up
+## 2. Design
 
-## Design & Implementation
+{Describe the design of the product. Use **Architecture Diagram** which has not been covered yet.}
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 3. Implementation
+### 3.1 [Proposed] Features
+#### 3.1.1 List By Category
+It extends the current view command.
+The user can use `list` to view all tasks.
+The user can also use `list CATEGORY` to view specific classes whose category is sth. 
+This is the UML design for view by category.
+![UML for View](images/viewCategory.png)
 
+<p>
+Step1: Users add task and add either a default category/new category.
+If user adds default category, then the categoryMap in taskList will not be modified.
+If the user adds a new category, the categoryMap wil add one more key whose value is the category and the value is
+the index for this task.
+</p>
 
-## Product Scope
-### Target user profile
 
-{Describe the target user profile}
+Step2:  User view tasks by category.
+If user only type in `list`, then all tasks will be displayed according to the sequence of task
+index. It simply calls the view function.
+If user type in in `list CATEGORY`, and the category is valid, then it will go for the
+returnCategory method.
+If user type in in `list CATEGORY`, and the category is invalid, then it will go for the
+displayCategory method. 
+Whatever the path, the UI will finally be called either display tasks given the index or display
+category. 
 
-### Value proposition
+#### 3.1.2 [Proposed] View month
+##### 3.1.2.1 Proposed Implementation
+The view month mechanism is facilitated by CalendarCommand which extends Command.
 
-{Describe the value proposition: what problem does it solve?}
+Given below is an example usage scenario and how the mechanism behaves at each step.
 
-## User Stories
+1. The user inputs the command word `calendar`. Upon which, the instance of parser will return a CalendarCommand for execution
+1. CalendarCommand initialises with the following variables, with the help from a calendar class containing the necessary methods related to day/date.
```

Comment:
I have the impression that the words *parser* in line 54, and *calendar* in line 55 refer to the names of classes used in the code. Will it be better to capitalize the first letter of their names, e.g. *Parser* instead of *parser*, so that it will be more consistent with your other class names such as *CalendarCommand* in line 55?